baybe/recommenders/meta/sequential.py

```diff
@@ -249,13 +252,13 @@ def select_recommender(  # noqa: D102
         # Remember the training dataset size for the next call
         self._n_last_measurements = len(train_x)
 
-        # FIXME: enable predictive recommenders for empty training data
+        # TODO: enable bayesian recommenders for empty training data
         if (train_x is None or len(train_x) == 0) and not isinstance(
             self._last_recommender, NonPredictiveRecommender
         ):
             raise _unsupported_recommender_error
 
-        return self._last_recommender
+        return self._last_recommender  # type: ignore[return-value]
```

Comment:
Just to make sure I understand this correctly: MyPy currently complains because this function always returns a `NonPredictiveRecommender`, although the type hint specifies `PureRecommender`? Or what is the issue here?

Reply:
i think it is because `last_recommender` is specified in the class as `Optional[PureRecommender]`, hence it thinks it could be `None` here

It does not realize here that the logic above avoids that

this can better be fixed once we dont have to distinguish anymore between Bayesian and NonPredictive ie #130  so I added the exception here